cf-agent/nfs.c

```diff
@@ -667,13 +667,26 @@ PromiseResult VerifyUnmount(EvalContext *ctx, char *name, Attributes a, const Pr
 static int MatchFSInFstab(char *match)
 {
     Item *ip;
+    const char delimit[]=" \t\r\n\v\f";
+    char *fstab_line, *token;
 
     for (ip = FSTABLIST; ip != NULL; ip = ip->next)
     {
-        if (strstr(ip->name, match))
+        fstab_line = xstrdup(ip->name);
+        if(strncmp(fstab_line, "#", 1) != 0)
```

Comment:
@skreuzer I think this is not ideal either, it will miss `<SPACE>#comment`

Please consider submitting another change moving the code to regular expressions and refactoring as @jimis seconded. If you don't plan to, let us know please.